boards/arm/stm32mp157c_dk2/doc/stm32mp157_dk2.rst

```diff
@@ -0,0 +1,302 @@
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig``
+
+
+Connections and IOs
+===================
+
+STM32MP157C-DK2 Discovery Board schematic is available here: `STM32MP157C Discovery board schematics`_.
+
+
+System Clock
+------------
+
+The Cortex®-M4 Core is configured to run at a 209 MHz clock speed.
+
+Serial Port
+-----------
+
+The STM32MP157C-DK2 Discovery board has 8 U(S)ARTs.
+The Zephyr console output is assigned by default to the ram console to be dumped
+by the Linux Remoteproc Framework on Cortex®-A7 core. The UART 4 can enabled as
+Cortex®-M4 console.
+
+Programming and Debugging
+*************************
+The STM32MP157C doesn't have QSPI flash for the Cortex®-M4  and it needs to be
+started by the Cortex®-A7 core. The Cortex®-A7 core is responsible to load the
+Cortex®-M4 binary application into the RAM, get the Cortex®-M4 out of reset.
+The Cortex®-A7 can perform these steps at bootloader level or after the Linux
+system has booted.
+
+The Cortex®-M4 can use up to 2 different RAMs. The program pointer starts at
+address 0x00000000 (RETRAM), the vector table should be loaded at this address
+These are the memory mapping for Cortex®-A7 and Cortex®-M4:
+
++------------+-----------------------+------------------------+----------------------+
+| Region     | Cortex®-A7            | Cortex®-M4             | Size                 |
++============+=======================+========================+======================+
+| RETRAM     | 0x38000000-0x3800FFFF | 0x00000000-0x0000FFFF  | 64KB                 |
++------------+-----------------------+------------------------+----------------------+
+| MCUSRAM    | 0x10000000-0x1005FFFF | 0x10000000-0x1005FFFF  | 384KB                |
++------------+-----------------------+------------------------+----------------------+
+| DDR        | 0xC0000000-0xFFFFFFFF |                        | up to 1 GB           |
++------------+-----------------------+------------------------+----------------------+
+
+
+Please refer to `stm32mp157c boot Cortex-M4 firmware`_ wiki page for instruction to load and
+start the Cortex-M4 firmware.
+
+Debugging
+=========
+
+You can debug an application using OpenOCD and GDB. The Solution proposed below
+is based on the Linux STM32MP1 SDK OpenOCD and is available only for Linux environment.
+The firmware have first to be loaded by Cortex®-A7. Then to debug the zephyr
```

Comment:
I'm not sure what "attach on it" means, and this rewrite might not be technically correct:
```
The firmware must first be loaded by the Cortex®-A7. The board must be attached
to your development system to debug the Zephyr application using OpenOCD
via the serial port.
```

Reply:
Attach means here that developer can connect a debugger on the fly to debug the software that as been already loaded and is running. As there is some interactions between the Zephyr and the Linux ( e.g rpmsg) this mode is privileged to not reset the Cortex-M4 with the debugger.

Proposal: `The firmware must first be loaded by the Cortex®-A7. Then developer can attach the debugger to the running Zephyr using OpenOCD.`